eclipse-collections-code-generator/src/main/resources/api/primitiveIterable.stg

```diff
@@ -127,6 +127,108 @@ public interface <name>Iterable extends PrimitiveIterable
         }
     }
 
+    /**
+     * Returns true if any of the values specified in the source array are contained
+     * in the <name>Iterable, and false if they are not.
+     *
+     * @since 11.0
+     */
+    default boolean containsAny(<type>... source)
+    {
+        <name>Iterable inside = this;
```

Comment:
Sorry, I just re-read this and I think it needs some adjustment, specifically, when `source.length == 32` (or bigger than maybe 2, less than 33). In those cases, the logic should essentially be like `Iterable` case. Can we write an acceptance test with source.length==32, this.size() == 1M, not a set (as a rough benchmark)? I think the following would do better:
```java
if (source.length \< 3 || this instanceof <name>Set) || this.size() \< 32) 
{
        for (int i = 0; i \< source.length; i++)
        {
            if (this.contains(source[i]))
            {
                return true;
            }
        }
        return false;
}
return this.containsAny(Arrays.asList(source)); // we have to create some garbage in this case, so let's not worry about the small wrapper
```

Reply:
No worries!

Added acceptance tests for containsAny and containsNone with source.length == 32 and this.size() == 1M. Runtime of those on my Macbook is ~50ms with the current implementation.

Tried changing it to what you provided above but containsAny can't take the output of Arrays.asList() and gives me the following error:

```
error: no suitable method found for containsAny(List<int[]>)
    method IntIterable.containsAny(int...) is not applicable
      (varargs mismatch; no instance(s) of type variable(s) T exist so that List<T> conforms to int)
    method IntIterable.containsAny(IntIterable) is not applicable
      (argument mismatch; no instance(s) of type variable(s) T exist so that List<T> conforms to IntIterable)
  where T is a type-variable:
    T extends Object declared in method <T>asList(T...)
```

I also can't use either IntSets or IntLists to make a primitive iterable since these are default methods in org.eclipse.collections.api and those are in org.eclipse.collections.impl. 🤷 Not sure where to go from here but I did push the acceptance tests I added.

Reply:
You can’t use the factories in API because of the packages and primitives are not onboarded to classloaders like the object side. You cannot use Arrays.asList() as you need to remain in primitive collections. 
An option which you have is to move this optimization down to implementations where you can use the factories and keep the unoptimized default method in the interface.

Reply:
before we go down that route, let's figure out the performance impact. Run the proposed new code and compare the timing with the older one.

Reply:
The older (current) code completes containsAny and containsNone with source of size 32 and the Iterable of size 1M in 50-60ms on my macbook.

To try out the proposed new code I overrode `containsAny(int... source)` and `containsNone(int.. source)` on AbstractIntIterable with it. I did change `Arrays.asList(source)` to `IntLists.mutable.with(source)` since I couldn't use Arrays.asList there.

With the same acceptance test (size.length == 32 and this.size() == 1M), the proposed new code completes containsAny and containsNone in 100-120ms on my macbook. It actually seems slower.

Reply:
ok, great. We're done then.